pkg/controllers/dynakube/oneagent/daemonset/arguments.go

```diff
@@ -13,7 +13,11 @@ const customArgumentPriority = 2
 const defaultArgumentPriority = 1
 
 func (dsInfo *builderInfo) arguments() ([]string, error) {
-	argMap := prioritymap.New(prioritymap.WithSeparator(prioritymap.DefaultSeparator), prioritymap.WithPriority(defaultArgumentPriority))
+	argMap := prioritymap.New(
```

Comment:
Why do we need a priorityMap at all? Wouldn't it be simpler to just append the args as they are? And not store them in proprietary struct

Reply:
Maybe, but not in this PR.

Reply:
The priority map ensures proper ordering of the arguments, no matter in which order they are added. IMHO this also provides more resilience against inconsiderate changes in the future.

Reply:
I am fine with not having it in this bugfix PR, but we should definitely get rid of this logic, its to complicated for what it brings us in terms of benefits